contracts/token/ERC20Channels.sol

```diff
@@ -0,0 +1,303 @@
+pragma solidity ^0.4.18;
+
+import "./ERC20.sol";
+import "../math/SafeMath.sol";
+import "../ECRecovery.sol";
+
+/**
+   @title ERC20Channels, State channels for ERC20 Tokens
+
+   Contract that provides holders of a ERC20 compatible token the creation of
+   channels between two users, once a channel is open the users can exchange
+   signatures offchain to agree on the final value of the transfer.
+   Uses OpenZeppelin ERC20 and SafeMath lib.
+ */
+contract ERC20Channels {
+  using SafeMath for uint256;
+
+  // Add recover method for bytes32 using ECRecovery lib from OpenZeppelin
+  using ECRecovery for bytes32;
+
+  // The ERC20 token to be used
+  ERC20 public token;
+
+  // The amount of time that a receiver has to challenge the sender
+  uint256 public challengeTime;
+
+  // The channels opened
+  mapping (bytes32 => Channel) public channels;
+  struct Channel {
+    uint256 deposit;
+    uint8 nonce;
+  }
+
+  // The requests from sender to close a channel
+  mapping (bytes32 => ClosingRequest) public closingRequests;
+  struct ClosingRequest {
+    uint256 closingBalance;
+    uint8 state;
+    uint256 closeTime;
+  }
```

Comment:
I think this way the contract storage will be smaller, it will store only the information necessary at any stage, if it is open only the channel information and once it want to be closed it will also store a closeRequest. @spalladino do you know if working with the structs like this will really optimize the contract storage ? Maybe im wrong!

Reply:
As far as I understand, storing a value of zero in storage has no gas cost, so it should make no difference. But I could be wrong, I think it's worth testing.